client/my-sites/checkout/checkout/prepurchase-notices/cart-plan-overlaps-owned-product-notice.tsx

```diff
@@ -33,7 +33,7 @@ const CartPlanOverlapsOwnedProductNotice: FunctionComponent< Props > = ( {
 	const translate = useTranslate();
 	const purchases = useSelector( ( state ) => getSitePurchases( state, selectedSite?.ID ) );
 	const purchase = isArray( purchases )
-		? purchases.find( ( p ) => p.productSlug === product.productSlug )
+		? purchases.find( ( p ) => p && p.productSlug === product.productSlug )
```

Comment:
I think it's safe to remove this guard now because I'm nearly certain I was wrong about this being possible (due to the nature of `getSitePurchases`).